File: java/Question1.java
class Question1 {
	public static void main(String[] args) {
		double sum = 0;

		sum += sumOfMultiples(3,999);
		sum += sumOfMultiples(5,999);
		sum -= sumOfMultiples(15,999);
		System.out.println("Sum of the products is " + sum);		
	}

	private static double sumOfMultiples(double factor, double limit) {
		double lastMultiple = limit - (limit % factor);
		double numberOfMultiples = lastMultiple / factor;
		return (factor + lastMultiple) * (numberOfMultiples/2) ;
	}
}

File: Rust/Question1.rs
fn main() {
	
	fn sum_of_multiples(factor: f64, limit: f64) -> f64 {
		let last_multiple: f64 = limit - (limit % factor);
		let number_of_multiples: f64 = last_multiple / factor;
		(factor + last_multiple)*(number_of_multiples/2.0)
	}

	let sum_of_multiples_of_3 = sum_of_multiples(3.0, 999.0);
	let sum_of_multiples_of_5 = sum_of_multiples(5.0, 999.0);
	let sum_of_multiples_of_15 = sum_of_multiples(15.0, 999.0);
	
	

	let sum = sum_of_multiples_of_3 + sum_of_multiples_of_5 - sum_of_multiples_of_15;

	println!("Sum of products is {}",sum);
}